### PR TITLE
Delete proxy backend

### LINKED ISSUE #1509: Delete Proxy backend from API Umbrella
If user deletes an API from Apinf and it is connected to a/the proxy, the proxy backend needs to be deleted both from Apinf collections and from API Umbrella backends. This task is about deleting it from API Umbrella.

# Definition of done
* [x] Ensure that the API is removed through API Umbrella Admin API and will not exist in API Umbrella database.
* [x] Publish the changes in Apinf API Umbrella.

### PR DESCRIPTION
**Pull request by jykae**
Closes #1509 

## Proposed changes
- delete proxyBackend feature
- improve code style where appropriate with my new friend ESLint

### CODE REVIEW

**Comment by brylie** [comment]
The following submit/delete button group code works properly, where clicking delete does not trigger the update event:

```html
<div class="btn-group pull-left">
      <!-- submit button -->
      <button
        type="submit"
        class="btn btn-success"
        id="save-proxy-button">
        {{_ "proxyBackendForm_saveButton" }}
      </button>
      <!-- delete button -->
      <button
        type="button"
        class="btn btn-danger"
        id="delete-proxy-button">
        Delete
      </button>
    </div>
```

**Comment by brylie** [comment]
You will also need to change your **form.less**:

```css
/* Change the following */
#save-proxy-button {
  #margin-top: 1em;
}

/* to */
#proxy-form-buttons {
  #argin-top: 1em;
}
```

**Comment by jykae** [comment]
@brylie Ok, thanks. I try out. Just pushed functionality.

**Comment by jykae** [comment]
@brylie Seems to work. I push also UI changes.

**Comment by jykae** [comment]
@apinf/developers ready for review

File: proxy_backends/client/form/form.js [hunk 1]
```diff
@@ -111,3 +117,56 @@ Template.proxyBackend.helpers({
...
+    // Get template instance
+    const instance = Template.instance();
+
+    // Get proxyBackend from template data
+    const proxyBackend = instance.data.proxyBackend;
+
+    // Check proxyBackend exists
+    if (proxyBackend) {
+      // Check if proxyBackend type is apiUmbrella & it has id
+      if (proxyBackend.apiUmbrella && proxyBackend.apiUmbrella.id) {
```
Comment (brylie) [inline proxy_backends/client/form/form.js]: Unless there is a syntactic reason to separate these statements, combine them into one `if` statement. Separate each conditional on its own line:

```js
if (proxyBackend &&
 proxyBackend.apiUmbrella &&
 proxyBackend.apiUmbrella.id ) {
  // do something
}
```

File: proxy_backends/client/form/form.js [hunk 2]
```diff
@@ -111,3 +117,56 @@ Template.proxyBackend.helpers({
...
+        const umbrellaBackendId = instance.data.proxyBackend.apiUmbrella.id;
+
+        // Delete API Backend on API Umbrella
+        Meteor.call(
+          'deleteApiBackendOnApiUmbrella',
+          umbrellaBackendId,
+          (deleteError) => {
+            if (deleteError) {
+              const deleteErrorMessage = TAPi18n.__('proxyBackendForm_deleteErrorMessage');
+              sAlert.error(`${deleteErrorMessage}:\n ${deleteError}`);
```
Comment (brylie) [inline proxy_backends/client/form/form.js]: Nice use of template strings :smile:

**Comment by brylie** [comment]
This looks good. Try to reduce the amount of nesting involved in the delete event handler. That will make the code slightly easier to read.

**Pull request merged**